src/liballoc/boxed.rs

```diff
@@ -63,6 +63,27 @@
 //! T` obtained from `Box::<T>::into_raw` may be deallocated using the
 //! [`Global`] allocator with `Layout::for_value(&*value)`.
 //!
+//! `Box<T>` has the same ABI as `&mut T`. In particular, when `T: Sized`,
+//! this allows using `Box<T>` in FFI:
```

Comment:
Just to make sure this does not get lost: This should talk about alignment, say that a `Box` is always required to be aligned even when it is not used.

Reply:
Isn't each instance of a type required to be aligned according to its type? `Box<T>` must own a valid `T`. Doesn't that entail that the `T` instance must be properly aligned? The example code also does not explicitly indicate that the pointer passed to `bar` must have been returned by a previous call to `foo` or be null, and does not mention that passing some random memory address is also undefined. `Box::from_raw` does not mention alignment either.

If you think that it is important to state nonetheless, does that wording look right: "A `*mut T` pointer used as `Box<T>` is always required to be aligned, even when not dereferenced."? Thank you.

Reply:
Validity of `T` is defined in terms of which sequences of bytes form a valid `T`; there is no such thing as an "alignment" of  sequence of bytes. Alignment only comes up when you consider that sequence of bytes stored in memory.

But you are right that being aligned is not sufficient: For a `*mut T` to be legal to turn into a `Box<T>`, we need more conditions. We could link [to these docs](https://doc.rust-lang.org/nightly/std/boxed/struct.Box.html#method.from_raw) but they are not great either. But [this section](https://doc.rust-lang.org/nightly/std/boxed/index.html#memory-layout) sounds useful.

Reply:
Thank you for explaining. Please let me know if you would like me to add a link to this other section of the docs or anything else to this PR.

Reply:
Yeah I think a link to https://doc.rust-lang.org/nightly/std/boxed/index.html#memory-layout would be good.

Reply:
This PR actually appends documentation to this "Memory layout" section, so I think a (self referential) link is not needed.